src/main/scala/net/snowflake/spark/snowflake/pushdowns/querygeneration/MiscStatement.scala

```diff
@@ -168,8 +170,21 @@ private[querygeneration] object MiscStatement {
   private final def setToExpr(set: Set[Any]): Seq[Expression] = {
     set.map {
       case d: Decimal => Literal(d, DecimalType(d.precision, d.scale))
-      case s: String => Literal(s, StringType)
+      case s @ (_:String | _:UTF8String) => Literal(s, StringType)
       case e: Expression => e
+      case default =>
+        try {
+          Literal(default)
+        }
+        catch {
+          case _: RuntimeException =>
```

Comment:
Some comments:
1. It sounds it is unclear what will happen for "Literal(default)".
    I prefer to specific more CASE for different types.
2. Are their specific reason to use "RuntimeException" instead of Exception?

Reply:
1. The code for Literal(default) is here: https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/literals.scala#L56 
There are too many types, I don't want to copy them all into here. I thought about removing the previous cases and only use this default case, but that might introduce behavior change so I only added this new case. 
2. Yes using Exception to catch all error would be better.